src/index.tsx

```diff
@@ -433,25 +435,19 @@ const Toaster = (props: ToasterProps) => {
         setToasts((toasts) => toasts.map((t) => (t.id === toast.id ? { ...t, delete: true } : t)));
         return;
       }
+      setToasts((toasts) => {
+        const indexOfExistingToast = toasts.findIndex((t) => t.id === toast.id);
+
+        // Update the toast if it already exists
+        if (indexOfExistingToast !== -1) {
+          return [
+            ...toasts.slice(0, indexOfExistingToast),
+            { ...toasts[indexOfExistingToast], ...toast },
+            ...toasts.slice(indexOfExistingToast + 1),
+          ];
+        }
 
-      // Prevent batching, temp solution.
-      setTimeout(() => {
-        ReactDOM.flushSync(() => {
```

Comment:
This is necessary to ensure correct calculation when multiple toasts are added at the same time.

Reply:
I tested it out and was able to render multiple toasts at the same time with no issues.

Reply:
````
toast('My action toast', {
    action: {
		label: 'Action',
		onClick: () => console.log('Action!'),
	},
});
````


Just threw this inside of a button and spammed it

Reply:
Yea that’s not what I meant, you need to invoke 2 toasts with one click to experience the bug ‘flushSync’ solves

Reply:
Oh I see it now. Interesting... Okay. I will fix that back.

Reply:
I got it updated. I also added another testing button for rendering multiple toasts.